src/js/osweb/backends/log.js

```diff
@@ -7,109 +7,22 @@ export default class Log {
-  }
-
-  /**
-   * Retrieves a list of all variables that exist in the experiment.
-   * @return {Array} - A list of all variables.
-   */
-  _get_all_vars () {
-    // Retrieves a list of all variables that exist in the experiment.
-    if (this._all_vars === null) {
-      this._all_vars = this._experiment.vars.inspect()
-    }
-    return this._all_vars
-  }
-
-  /** Closes the current log. */
-  close () {
-    // Closes the current log.
-    if (this._log.length > 0) {
-      // Echo the data to the runner.
-      this._experiment._runner._data = this._log.join('')
-    };
-
-    // Clear the log file.
-    this._log = []
-  }
-
-  /** Opens the current log. If a log was already open, it is closed. */
-  open () {
-    // Opens the current log. If a log was already open, it is closed.
-    this._header_written = false
-
-    // Check for old data.
-    if (this._log !== null) {
-      // Clear the old data.
-      this.close()
-    }
-  }
-
-  /**
-   * Write one signle line to the message log.
-   * @param {String} msg - Message to add to the log file.
-   * @param {Boolean} newLine - If true a new line character is inserted into the message.
-   */
-  write (msg, newLine) {
-    // Write one message to the log.
-    newLine = (typeof newLine === 'undefined') ? true : newLine
-
-    if (newLine === true) {
-      // Write a log with a new line.
-      this._log.push(msg + '\n')
-    } else {
-      // Write a log without a new line.
-      this._log.push(msg)
-    }
   }
 
   /**
    * Write one signle line to the message log.
    * @param {Array} varList - Array with variables to write to the log.
    */
   write_vars (varList) {
-    // Writes variables to the log.
-    varList = (typeof varList === 'undefined') ? null : varList
-
-    var value
-    var l = []
-    // If no var list defined, retrieve all variables.
-    if (varList === null) {
-      varList = this._get_all_vars()
-    }
-
-    // Sort the var list.
-    varList.sort()
-
-    // Add the header to the log file.
-    if (this._header_written === false) {
-      for (var i = 0; i < varList.length; i++) {
-        l.push('"' + varList[i] + '"')
-      }
-      this.write(l.join())
-      this._header_written = true
-    }
-
-    // Add the data entries to the log file.
-    l = []
-    const entry = {}
-    for (let i = 0; i < varList.length; i++) {
-      value = this._experiment.vars.get(varList[i], 'NA', false)
+    if (!isFunction(this._experiment.onLog)) return
+    let entry = {}
+    let value
+    for (let varName of varList) {
+      value = this._experiment.vars.get(varName, 'NA', false)
       if (isFunction(value)) continue
-      l.push('"' + value + '"')
-      entry[varList[i]] = value
-    }
-    this.write(l.join())
-
-    // If event is attached to the experiment output log.
-    if (isFunction(this._experiment.onLog)) {
-      this._experiment.onLog(entry)
+      entry[varName] = value
     }
+    this._experiment.onLog(entry)

```

Comment:
the  default `onLog` function now always prints the entry to the console, which may be undesirable. Maybe it's better to change the `console.log` statement in this callback function to `this.debugger.addMessage`, or something that can be switched off by the context object with which osweb is initialized?

Reply:
Let's leave it for now, because it's better than *not* seeing the log at all. I was thinking that, for standalone mode, we might also save the log and then let the user download it at the end of the experiment. That way people might even use the standalone mode for actual testing. But that's not for this PR.